src/modules/ticket-fields/fields/LookupField.tsx

```diff
@@ -17,6 +17,12 @@ import debounce from "lodash.debounce";
 import { useTranslation } from "react-i18next";
 import { EmptyValueOption } from "./EmptyValueOption";
 import type { LookupRelationshipFieldFilter } from "../data-types/BaseTicketField";
+import { ASSET_TYPE_KEY } from "../../shared/asset-management/constants";
+import type { RawOption } from "../../service-catalog/components/service-catalog-item/ItemRequestForm";
+import type {
+  AssetsApiResponse,
+  AssignedAssetFields,
+} from "../../service-catalog/data-types/Assets";
```

Comment:
The `LookupField` is also used for in the new request form and in that context it doesn't make sense that we are adding `ITAM` specific logic to it.
This component must remain agnostic to Service Catalog and ITAM so it should not be importing anything from `service-catalog`.

As an example, this component works for any custom object and not just `assets`. We need to find a better name instead `applyAssetFiltersAsync`, e.g. `applyFilters`.